demos/tutorials/Gibbs.py

```diff
@@ -0,0 +1,222 @@
+"""
+Gibbs sampling
+==============
+
+    This tutorial shows how to use cuqipy to perform Gibbs sampling.
+    Gibbs sampling is a Markov chain Monte Carlo (MCMC) method for
+    sampling a joint probability distribution.
+
+    Opposed to jointly sampling the distribution simultaneously, Gibbs
+    sampling samples the variables of the distribution sequentially,
+    one variable at a time. The sampling of each variable is done by
+    sampling from the conditional distribution of that variable given 
+    the values of the other variables.
+
+    This is often a very efficient way of sampling from a joint
+    distribution if the conditional distributions are easy to sample
+    from. This is one way to exploit the structure of the joint
+    distribution.
+
```

Comment:
Maybe mention that it may be slow for correlated variables?

Reply:
Good suggestion